Feat/106/todo card modal component

rak517 opened the pull request:
## ❓이슈
- close #106 

## :writing_hand: Description

- 할 일 카드의 상세정보 모달 구현
- 상세정보의 댓글 기능 구현
![스크린샷 2025-02-13 16 42 08](https://github.com/user-attachments/assets/d56dc6eb-fed8-44ea-9ac7-ad796065243e)
![스크린샷 2025-02-13 16 44 02](https://github.com/user-attachments/assets/5dfb5eb0-1da7-4fd5-9fa2-17d8eae8ffa1)


## :white_check_mark: Checklist

### PR

<!-- 작성중인 PR인 경우, Draft 모드로 생성해주세요. -->

- [x] Branch Convention 확인
  > `feat/*` 기능 구현, `fix/*` 버그 수정, `refactor/*` 개선
- [x] Base Branch 확인
- [x] 적절한 Label 지정
- [x] Assignee 및 Reviewer 지정

### Test

- [x] 로컬 작동 확인

### Additional Notes

<!-- 추가 사항이 있을 경우, Todo list를 작성해주세요. -->

- [ ] (없음)

rak517 committed:
:sparkles: feat: 할일 카드 모달 생성

rak517 committed:
:sparkles: feat: 할 일 카드 상세정보 모달 구현

rak517 committed:
:bug: fix: 드롭다운, 이미지 수정

rak517 committed:
:fire: remove: 불필요한 console.log 삭제

rak517 committed:
:recycle: refactor: confirm 사용 부분 useConfirm으로 교체

ToKyun02 commented on src/components/dashboard/TodoCard.tsx:
```diff
@@ -1,37 +1,50 @@
+              <Image src={calendar} width={10} height={11} alt='캘린더' />
+              <span className='text-xs font-medium text-gray-50'>{formattedDate}</span>
+            </div>
+            <Avatar email={card.assignee.nickname} size='sm' />
```
Avatar 렌더링할 때 담당자의 profileImage도 prop으로 넘겨줘야 할 것 같습니다!

profileImageUrl = {card.assignee.profileImageUrl} 로 설정하시면 될 것 같아요

ToKyun02 commented on src/components/dashboard/TodoCard.tsx:
```diff
@@ -1,37 +1,50 @@
+        onClick={() => detailTodoModalRef.current?.open()}
+      >
+        {card.imageUrl && <Image src={card.imageUrl} alt={card.title} className='w-full md:w-[90px] lg:w-full' width={90} height={20} />}
+        <div className='flex flex-col gap-1.5 md:flex-1 md:flex-row lg:flex-col lg:gap-2'>
```
card의 이미지가 작아서 실제로 카드 렌더링되면 이미지 화질이 깨지는 이슈가 있습니다.

width랑 height를 더 줘야 할 것 같아요

ToKyun02 commented on src/components/dashboard/DetailTodo.tsx:
```diff
@@ -0,0 +1,159 @@
+      router.push('/dashboard');
+    } catch (err) {
+      alert('카드 정보를 불러오는 중 오류가 발생했습니다.');
+      console.error(err);
```
console.error는 빼도 될 것 같습니다!

ToKyun02 commented on src/components/dashboard/DetailTodo.tsx:
```diff
@@ -0,0 +1,159 @@
+  card: Card;
+}
+
+const NO_IMAGE_BASE_URL = 'https://sprint-fe-project.s3.ap-northeast-2.amazonaws.com/taskify/task_image/';
```
빈 이미지 등록은 제가 나중에 이미지 등록하고 URL 알려드리겠습니다!

ToKyun02 reviewed (commented):
작업하시느라 고생 많으셨습니다!

comment api 사용할 때 리액트 쿼리 사용해야 할 것 같습니다!
카드를 닫았다가 다시 들어오면 재 페칭되는걸 방지하는 목적도 있어용

지금은 아직 렌더링만 보는 단계인 것 같아서 추후 다른 API 사용할 때 적용해도 될 것 같습니다!

cards의 경우 제가 지금 작업 중인 이슈에서 쿼리 작성완료했습니다

cksrlcks committed:
Feat/40/mypage refactor (#126)

* ♻️ refactor : 이미지 업로드 개선
- 타입에 null 추가 (api에서 응답이 null이 올 수 있음)
- 쿼리 개선 (zod validation에 용량 추가)
- defaultValue 추가 (추가로 올린 이미지를 지우면, 원래 이미지가 보이게)

* ♻️ refactor : 내정보 페이지 개선
- 내정보를 ssr로 불러오도록 개선
- 폼내부 공용 컴포넌트로 교체
- 폼내부 로직 개선

* 🐛 fix : 헤더 내정보에 프로필 이미지 노출 추가

SeokChan-Lee committed:
🐛 fix: useParams로 방문중인 대시보드 표시 개선 (#128)

Co-authored-by: chanki.kim <chanki.kim89@gmail.com>

SeokChan-Lee committed:
:lipstick: style: 로그인,회원가입 페이지 로고 png->svg변경 (#125)

rak517 committed:
:recycle: refactor: 코드 리뷰 반영

rak517 committed:
:bug: fix: Avartor 컴포넌트 profileImageUrl 오류 해결
